src/TransitionGroup.js

```diff
@@ -37,12 +37,31 @@ const propTypes = {
    * on individual children Transitions.
    */
   exit: PropTypes.bool,
+
+  staggerTime: PropTypes.number,
 };
 
 const defaultProps = {
   component: 'div',
 };
 
+function childMapperFn(child, index) {
+  const { staggerTime } = this.props;
+  const onExited = () => this.handleExited(child.key);
+  const style = staggerTime > 0 ? {
+    transitionDelay: `${staggerTime * (index + 1)}ms`
```

Comment:
Ah, I see your point. I'll see if I can re-work this without CSS. Thanks for the quick response.